### PR TITLE
Creates a new `schema` endpoint that returns the ImageStats schema

### PR DESCRIPTION
**Pull request by zakkak**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,6 +55,11 @@
             <groupId>io.quarkus</groupId>
             <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
         </dependency>
+        <dependency>
+            <groupId>com.fasterxml.jackson.module</groupId>
+            <artifactId>jackson-module-jsonSchema</artifactId>
+            <version>2.17.1</version>
```
Comment (Karm) [inline pom.xml:61]: @zakkak Could we move the version to `<properties>`, please?
Also, isn't this pulled already by Quarkus...?

File: src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java [hunk 2]
```diff
@@ -340,4 +344,20 @@ public Response toResponse(Exception exception) {
...
+    @RolesAllowed("token_read")
+    @Path("/schema")
+    @Produces(MediaType.APPLICATION_JSON)
+    public String schema() {
+        ObjectMapper mapper = new ObjectMapper();
+
+        JsonSchemaGenerator generator = new JsonSchemaGenerator(mapper);
+        try {
+            JsonSchema schema = generator.generateSchema(ImageStats.class);
+            return mapper.writerWithDefaultPrettyPrinter().writeValueAsString(schema);
```
Comment (Karm) [inline src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java:358]: Pretty printing by default? So it's for humans to see, right?

**Review by Karm — CHANGES_REQUESTED** [review]
THX @zakkak  :pray: 

I would like to see a simple test added, just a rudimentary thing to make sure the Jackson jsonSchema plays ball in native mode...

File: pom.xml (hunk 1 above)
Comment (zakkak) [reply pom.xml:61]: > @zakkak Could we move the version to `<properties>`, please? 

On it.

> Also, isn't this pulled already by Quarkus...?

Surprisingly, no :)

File: src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java (hunk 2 above)
Comment (zakkak) [reply src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java:358]: My use case was to copy it in order to feed it in another tool (Horreum) for validation of the input.

But I can see it being used mostly by humans, either to understand the schema or to copy paste it in some other place.

File: pom.xml (hunk 1 above)
Comment (zakkak) [reply pom.xml]: Done

**Comment by zakkak** [comment]
> I would like to see a simple test added, just a rudimentary thing to make sure the Jackson jsonSchema plays ball in native mode...

Test added. Note however that trying to run the tests in native mode with `./mvnw -Dnative verify -Dquarkus.native.container-runtime=podman` fails (not on the new test).

**Review by Karm — CHANGES_REQUESTED** [review]
@zakkak Native does not fail with current main, it just needs ` -Dquarkus.profile=test`
This PR fails on the schema test though.

File: src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java (hunk 2 above)
Comment (zakkak) [reply src/main/java/com/redhat/quarkus/mandrel/collector/report/endpoints/ImageStatsResource.java]: I ended up making it return the unformatted schema. One can easily pipe it to `jq` to pretty print.

**Review by Karm — APPROVED** [review]
Thx! Looks good now.

**Pull request merged**